sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -341,7 +341,7 @@ def mark_upstream_ios_of_output(template_output, marked_inputs,
     # Remove pipeline task parameters unless they're used downstream
     for task in pipeline_tasks:
         # Don't process condition and custom task parameters
-        if 'condition-' not in task['name'] and not task.get('taskRef', ''):
+        if 'condition-' not in task['name'] and not task.get('taskRef', '') and not task.get('taskSpec', ''):
```

Comment:
`not task.get('taskSpec', '')` will also include regular tasks in this case, so what we need to do here is not to process a task that has taskSpec with custom task. Unfortunately in Tekton yaml, there's no concept of API type like in goLang, so we probably need to check some special annotations (e.g. `valid_container`) to determine whether the taskSpec is for custom task.

Reply:
Oh, I see! Ok, so I'll factor these conditions out and add looking for `'apiVersion'` --- regular tasks don't have it while custom tasks always do.